packages/story-editor/src/app/mediaUpload/useMediaUpload.js

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { useContextSelector, identity } from '@web-stories-wp/react';
+/**
+ * Internal dependencies
+ */
+import Context from './context';
+
+function useMediaUpload(selector) {
+  const context = useContextSelector(Context, selector ?? identity);
+
+  if (!context) {
+    throw new Error(
+      'Must use `useMediaUpload()` within <mediaUpload.Provider />'
+    );
+  }
```

Comment:
I guess within this context, this is fine. But watch out for this pattern with `useContextSelector` because it's the context after it's been selected. So if say you have:
```
const someBool = useSomeContext((context) => context.state.someBool);
```
and someBool happens to be `false`, we wouldn't want an error thrown. 

Anyway, just giving a heads up cus I've ran into this before